test/e2e/tests/firefox-infobar-ui/translation.simultaneous-tabs.spec.ts

```diff
@@ -0,0 +1,67 @@
+/* eslint-env node, mocha */
+
+import { installExtension, launchFirefox } from "../../utils/setupWebdriver";
+import { navigateToURL } from "../../utils/navigateToURL";
+import { takeScreenshot } from "../../utils/takeScreenshot";
+import { translateViaInfobar } from "../../utils/translationInfobar";
+import { startTestServers } from "../../utils/setupServers";
+import { maxToleratedTelemetryUploadingDurationInSeconds } from "../../utils/telemetry";
+import {
+  assertOriginalPageElementExists,
+  assertTranslationSucceeded,
+  fixtureUrl,
+  maxToleratedModelLoadingDurationInSeconds,
+  maxToleratedTranslationDurationInSeconds,
+} from "../../utils/translationAssertions";
+
+let shutdownTestServers;
+
+let tabsCurrentlyOpened = 0;
+
+describe("Translation: Simultaneous tabs", function() {
+  // This gives Firefox time to start, and us a bit longer during some of the tests.
+  this.timeout(
+    (15 +
+      maxToleratedTelemetryUploadingDurationInSeconds +
+      maxToleratedModelLoadingDurationInSeconds +
+      maxToleratedTranslationDurationInSeconds * 5) *
+      1000,
+  );
+
+  let driver;
+
+  before(async function() {
+    const _ = await startTestServers();
+    shutdownTestServers = _.shutdownTestServers;
+    driver = await launchFirefox();
+    await installExtension(driver);
+    // Allow our extension some time to set up the initial ui
+    await driver.sleep(1000);
+  });
+
+  after(async function() {
+    await takeScreenshot(driver, this.test.fullTitle());
+    await driver.quit();
+    shutdownTestServers();
+  });
+
+  it("Translation via the infobar works when translating in multiple tabs at the same time", async function() {
```

Comment:
do we test that telemetry is sent properly when translating different language pairs on two tabs at the same time?

Reply:
No, that is a good point. Probably worthwhile adding...

Reply:
Added telemetry checks in e2e tests translation.simultaneous-tabs.spec.ts and translation.subsequent-navigations.spec.ts

Reply:
I don't think it verifies #49. Language pair is hardcoded in assertion but they should be different for different tabs

Reply:
Ah yes, true!